integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java

```diff
@@ -5,152 +5,185 @@
+    findSparkConfigKey(conf, SPARK_CONF_PARENT_RUN_ID).ifPresent(builder::parentRunId);
+    builder.openLineageYaml(extractOpenlineageConfFromSparkConf(conf));
+    return builder.build();
   }
 
-  private static Optional<String> getNamedStringParameter(
-      List<NameValuePair> nameValuePairList, String name) {
-    return Optional.ofNullable(getNamedParameter(nameValuePairList, name))
-        .filter(StringUtils::isNoneBlank);
+  // adjust properties so the old pipelines are allowed
+  private static void adjustOldConfigs(SparkConf conf) {
+    findSparkConfigKey(conf, "spark.openlineage.host")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(conf, SPARK_CONF_HTTP_URL, c, "spark.openlineage.host");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.timeout")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(
+                  conf, UrlParser.SPARK_CONF_TIMEOUT, c, "spark.openlineage.timeout");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.version")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(
+                  conf,
+                  UrlParser.SPARK_CONF_API_ENDPOINT,
+                  String.format("api/v%s/lineage", c),
+                  "spark.openlineage.version");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.apiKey")
+        .ifPresent(
+            c -> {
+              conf.setIfMissing(UrlParser.SPARK_CONF_API_KEY, c);
+              replaceConfigEntry(
+                  conf, UrlParser.SPARK_CONF_AUTH_TYPE, "api_key", "spark.openlineage.apiKey");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.consoleTransport")
+        .ifPresent(
+            c -> {
+              if (c.equals("true")) {
+                conf.set(SPARK_CONF_TRANSPORT_TYPE, "console");
+                conf.remove("spark.openlineage.consoleTransport");
+              }
+            });
+    Arrays.stream(conf.getAllWithPrefix("spark.openlineage.url.param."))
+        .forEach(
+            c -> {
+              conf.set("spark.openlineage.transport.urlParams." + c._1, c._2);
+              conf.remove("spark.openlineage.url.param." + c._1);
+            });
   }
 
-  private static Optional<Double> getTimeout(List<NameValuePair> nameValuePairList) {
-    return Optional.ofNullable(
-        ArgumentParser.extractTimeout(getNamedParameter(nameValuePairList, "timeout")));
+  private static void replaceConfigEntry(
+      SparkConf conf, String sparkConfHttpUrl, String c, String key) {
+    conf.setIfMissing(sparkConfHttpUrl, c);
+    conf.remove(key);
   }
 
-  private static Double extractTimeout(String timeoutString) {
-    try {
-      if (StringUtils.isNotBlank(timeoutString)) {
-        return Double.parseDouble(timeoutString);
+  public static OpenLineageYaml extractOpenlineageConfFromSparkConf(SparkConf conf) {
```

Comment:
wouldn't it be nice to cover it with some unit tests?

Reply:
[integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java](https://github.com/OpenLineage/OpenLineage/blob/418cfb235547655351d72ab2929c96df23e71768/integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java#L132)
here is example of test, already implemented